src/python/BFinitPassthrough.py

```diff
@@ -54,19 +58,21 @@ def bf_passthrough_init(port, requestedBaudrate, half_duplex=False):
         raise PassthroughEnabled("No CLI available. Already in passthrough mode?, If this fails reboot FC and try again!")
 
     serial_check = []
-    if not _validate_serialrx(rl, "provider", [["CRSF", "ELRS"], "GHST"][half_duplex]):
-        serial_check.append("serialrx_provider != CRSF")
-    if not _validate_serialrx(rl, "inverted", "OFF"):
-        serial_check.append("serialrx_inverted != OFF")
-    if not _validate_serialrx(rl, "halfduplex", ["OFF", "AUTO"]):
-        serial_check.append("serialrx_halfduplex != OFF/AUTO")
+    if _validate_serialrx(rl, "expresslrs_model_id", [str(x) for x in range(0,255)] ):
+        serial_check.append("ExpressLRS SPI RX detected\n\nUpdate via betaflight to flash your RX\nhttps://www.expresslrs.org/2.0/hardware/spi-receivers/")
```

Comment:
This detects a regular serial RX as well!

Reply:
I tested with a regular rx I didn’t think this command exists without an expresslrs spi rx!  But it wasn’t tested against 4.3rc2. If that is the case then it makes sense to change it.

Reply:
more info
```
Entering CLI Mode, type 'exit' to return, or 'help'

# 
# Building AutoComplete Cache ... Done!
# 
# version
# Betaflight / STM32F405 (S405) 4.3.0 Feb  1 2022 / 20:13:01 (3267f0417) MSP API: 1.44
# config: manufacturer_id: CLRA, board_name: CLRACINGF4, version: 14929e39, date: 2020-08-31T10:05:16Z
# board: manufacturer_id: CLRA, board_name: CLRACINGF4

# get express
expresslrs_uid = 0,0,0,0,0,0
Array length: 6

expresslrs_domain = AU433
Allowed values: AU433, AU915, EU433, EU868, IN866, FCC915, ISM2400

expresslrs_rate_index = 0
Allowed range: 0 - 3

expresslrs_switch_mode = HYBRID
Allowed values: HYBRID, WIDE

expresslrs_model_id = 255
Allowed range: 0 - 255
```

Reply:
Ok, that makes sense... And serial rx can be configured even with elrs spi package. Need to be fixed 👍

Reply:
On the H7RF I get this information
```
# version
# Betaflight / SPRACINGH7RF (SP7R) 4.3.0 Dec 20 2021 / 15:23:58 (711159dcbc) MSP API: 1.44

# get serialrx
serialrx_provider = SPEK1024
Allowed values: SPEK1024, SPEK2048, SBUS, SUMD, SUMH, XB-B, XB-B-RJ01, IBUS, JETIEXBUS, CRSF, SRXL, CUSTOM, FPORT, SRXL2, GHST

serialrx_inverted = OFF
Allowed values: OFF, ON

serialrx_halfduplex = OFF
Allowed values: OFF, ON

# get rx_spi
rx_spi_protocol = EXPRESSLRS
Allowed values: V202_250K, V202_1M, SYMA_X, SYMA_X5C, CX10, CX10A, H8_3D, INAV, FRSKY_D, FRSKY_X, FLYSKY, FLYSKY_2A, KN, SFHSS, SPEKTRUM, FRSKY_X_LBT, REDPINE, FRSKY_X_V2, FRSKY_X_LBT_V2, EXPRESSLRS

rx_spi_bus = 2
Allowed range: 0 - 6

rx_spi_led_inversion = OFF
Allowed values: OFF, ON
```

Reply:
Ah I think it’s because this board has exposed spi pads right?

Reply:
I think H7RF has still its own codebase. @hydra to confirm this.
We should follow official versions only...

Reply:
The H7RF is built from a rebased BF `master` branch with no changes to the ELRS code other than the addition of an LQ graph OSD Element.

See https://github.com/betaflight/betaflight/pull/11110

Latest branches here:

https://github.com/spracing/betaflight/branches/all?query=h7rf

Reply:
I think this is the best way to handle this.

diff --git a/src/python/BFinitPassthrough.py b/src/python/BFinitPassthrough.py
index 2d49047..5b51ad6 100644
--- a/src/python/BFinitPassthrough.py
+++ b/src/python/BFinitPassthrough.py
@@ -58,15 +58,14 @@ def bf_passthrough_init(port, requestedBaudrate, half_duplex=False):
         raise PassthroughEnabled("No CLI available. Already in passthrough mode?, If this fails reboot FC and try again!")
 
     serial_check = []
-    if _validate_serialrx(rl, "expresslrs_model_id", [str(x) for x in range(0,255)] ):
-        serial_check.append("ExpressLRS SPI RX detected\n\nUpdate via betaflight to flash your RX\nhttps://www.expresslrs.org/2.0/hardware/spi-receivers/")
-    else:
-        if not _validate_serialrx(rl, "serialrx_provider", [["CRSF", "ELRS"], "GHST"][half_duplex]):
-            serial_check.append("Serial Receiver Protocol is not set to CRSF! Hint: set serialrx_provider = CRSF")
-        if not _validate_serialrx(rl, "serialrx_inverted", "OFF"):
-            serial_check.append("Serial Receiver UART is inverted! Hint: set serialrx_inverted = OFF")
-        if not _validate_serialrx(rl, "serialrx_halfduplex", ["OFF", "AUTO"]):
-            serial_check.append("Serial Receiver UART is not in full duplex! Hint: set serialrx_halfduplex = OFF")
+    if not _validate_serialrx(rl, "serialrx_provider", [["CRSF", "ELRS"], "GHST"][half_duplex]):
+        serial_check.append("Serial Receiver Protocol is not set to CRSF! Hint: set serialrx_provider = CRSF")
+    if not _validate_serialrx(rl, "serialrx_inverted", "OFF"):
+        serial_check.append("Serial Receiver UART is inverted! Hint: set serialrx_inverted = OFF")
+    if not _validate_serialrx(rl, "serialrx_halfduplex", ["OFF", "AUTO"]):
+        serial_check.append("Serial Receiver UART is not in full duplex! Hint: set serialrx_halfduplex = OFF")
+    if _validate_serialrx(rl, "rx_spi_protocol", "EXPRESSLRS" ) and serial_check:
+        serial_check = [ "ExpressLRS SPI RX detected\n\nUpdate via betaflight to flash your RX\nhttps://www.expresslrs.org/2.0/hardware/spi-receivers/" ]
 
     if serial_check:
         error = "\n\n [ERROR] Invalid serial RX configuration detected:\n"
@@ -75,6 +74,9 @@ def bf_passthrough_init(port, requestedBaudrate, half_duplex=False):
         error += "\n    Please change the configuration and try again!\n"
         raise PassthroughFailed(error)
 
+
+
+
     SerialRXindex = ""
 
     dbg_print("\nAttempting to detect FC UART configuration...")

Reply:
Yes, those changes are whats needed. I validated that the correct things happen on both a serial connected RX and on the H7RF SPI RX.

Reply:
Thanks @pkendall64 and @mavericm1 !
I just committed the change.